pkg/asset/imagebased/configimage/clusterconfiguration.go

```diff
@@ -0,0 +1,163 @@
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ClusterConfiguration) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.ClusterID{},
+		&tls.KubeAPIServerLBSignerCertKey{},
+		&tls.KubeAPIServerLocalhostSignerCertKey{},
+		&tls.KubeAPIServerServiceNetworkSignerCertKey{},
+		&tls.AdminKubeConfigSignerCertKey{},
+		&password.KubeadminPassword{},
+		&agent.OptionalInstallConfig{},
+		&ImageBasedConfig{},
+	}
+}
+
+// Generate generates the Image-based Installer ClusterConfiguration manifest.
+func (cc *ClusterConfiguration) Generate(dependencies asset.Parents) error {
+	clusterID := &installconfig.ClusterID{}
+	installConfig := &agent.OptionalInstallConfig{}
+	imageBasedConfig := &ImageBasedConfig{}
+	serverLBSignerCertKey := &tls.KubeAPIServerLBSignerCertKey{}
+	serverLocalhostSignerCertKey := &tls.KubeAPIServerLocalhostSignerCertKey{}
+	serverServiceNetworkSignerCertKey := &tls.KubeAPIServerServiceNetworkSignerCertKey{}
+	adminKubeConfigSignerCertKey := &tls.AdminKubeConfigSignerCertKey{}
+
+	dependencies.Get(
+		clusterID,
+		installConfig,
+		imageBasedConfig,
+		serverLBSignerCertKey,
+		serverLocalhostSignerCertKey,
+		serverServiceNetworkSignerCertKey,
+		adminKubeConfigSignerCertKey,
+	)
+
+	pwd := &password.KubeadminPassword{}
+	dependencies.Get(pwd)
+	pwdHash := string(pwd.PasswordHash)
+
+	if installConfig.Config == nil || imageBasedConfig.Config == nil {
+		return cc.finish()
+	}
+
+	// TODO: add RawNMStateConfig
+	cc.Config = &lca.SeedReconfiguration{
+		// TODO: uncomment when https://github.com/openshift-kni/lifecycle-agent/pull/499 is merged
+		// AdditionalTrustBundle: (lca.PEM)installConfig.Config.AdditionalTrustBundle,
+		APIVersion:            lca.SeedReconfigurationVersion,
+		BaseDomain:            installConfig.Config.BaseDomain,
+		ClusterID:             clusterID.UUID,
+		ClusterName:           installConfig.ClusterName(),
+		Hostname:              imageBasedConfig.Config.Hostname,
+		InfraID:               clusterID.InfraID,
+		KubeadminPasswordHash: pwdHash,
+		Proxy:                 (*lca.Proxy)(installConfig.Config.Proxy),
+		PullSecret:            installConfig.Config.PullSecret,
+		ReleaseRegistry:       imageBasedConfig.Config.ReleaseRegistry,
+		SSHKey:                installConfig.Config.SSHKey,
+	}
+
+	cc.Config.KubeconfigCryptoRetention = lca.KubeConfigCryptoRetention{
+		KubeAPICrypto: lca.KubeAPICrypto{
+			ServingCrypto: lca.ServingCrypto{
+				LoadbalancerSignerPrivateKey:   (lca.PEM)(string(serverLBSignerCertKey.Key())),
+				LocalhostSignerPrivateKey:      (lca.PEM)(string(serverLocalhostSignerCertKey.Key())),
+				ServiceNetworkSignerPrivateKey: (lca.PEM)(string(serverServiceNetworkSignerCertKey.Key())),
+			},
+			ClientAuthCrypto: lca.ClientAuthCrypto{
+				AdminCACertificate: (lca.PEM)(string(adminKubeConfigSignerCertKey.Cert())),
+			},
+		},
+		// TODO: generate and add ingress CA key
```

Comment:
In a regular OCP installation, you won't have the ingress key at this point since it gets generated later during the cluster installation.
However, for IBI you can/should generate the cert and provide it in the config.
I'm not sure but I think that IBI installation will fail if you don't provide the ingress key here.

Reply:
Yes, you're exactly right. I left it as a `TODO` here because I  have to dive in the respective installer part(s) and make sure we generate and add the correct/proper key here in the `cluster-configuration` for IBI. 

I'll come back to you if I have issues implementing this :+1: